### PR TITLE
Update Flare to 1.14

### PR DESCRIPTION
**Pull request by Peppersawce**
I just took the old recipe and tweaked it a bit to see how it would go and it compiled without issues. It knows the right paths for config & userdata out of the box as well. The only actual "issue" it has is that it starts using the "SDL hardware" renderer by default which is way slower on Haiku than the plain SDL one.

I also decided to merge the flare-data package into the main one as it was considered a requirement by the main package anyway (although the core engine does start without the game data... and doesn't do much else).

Btw this merging meant I had to tweak the download links a bit so that Haiku's mirror sources would have different names and not overlap.

### CODE REVIEW

**Comment by Peppersawce** [comment]
Yeah, I think so, I'll experiment a little bit with it later, maybe there's a cmake value for it as well or something

**Comment by Peppersawce** [comment]
So, I wasn't able to change the default renderer in any way so I started experimenting a bit with making the game packaging more modular, in case that's a good idea. This alternate recipe (`flare_modular-1.14`) currently builds:

A base package with the engine, Alpha demo as the gameplay mod, then fantasycore & misc
A minicore package with the Minicore mods (to play on low resolutions) and gcw0_defaults (which may help if you want to run this on a handheld)
An empyrean package with the Empyrean Campaign, the official Flare campaign
Then polymorphable and nonamemod packages for 2 community-made Flare gameplay mods

Would a recipe like make or would it be better if I split the 2 community mods into their own separate recipe? Or should I scrap the idea altogether and use the old recipe instead? I guess most people would want the empyrean mod but not a lot of people actually need the stuff in minicore.

**Comment by korli** [comment]
The idea of a separate flare_data package, is that it avoids packing stale game data for each new revision of the game recipe.
How big is the flare package after your changes?

**Comment by Peppersawce** [comment]
I'm not sure I understand that last post, I didn't get rid of the flare data package, I only got rid of the separate recipe for it, so... the base game package is around 5 megs iirc?

File: games-rpg/flare/flare-1.14.recipe [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+COPYRIGHT="2010-2022 Clint Bellanger & Flare team"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/flareteam/flare-engine/archive/v$portVersion/flare-engine-v$portVersion.tar.gz"
+CHECKSUM_SHA256="2c1bafeaa5ac26c10449bfcb7151b06e8a22547aa7364d2a39bbcb135aa50a09"
+SOURCE_DIR="flare-engine-$portVersion"
+SOURCE_URI_2="https://github.com/flareteam/flare-game/archive/v$portVersion/flare-game-v$portVersion.tar.gz"
+CHECKSUM_SHA256_2="65758462070aa88842084f8ee69083d8226e93cfbf83481663276d8307494b8e"
+SOURCE_DIR_2="flare-game-$portVersion"
+ADDITIONAL_FILES="flare.rdef.in"
```
Comment (korli) [inline games-rpg/flare/flare-1.14.recipe:21]: Why did you remove the old patchset?

Comment (Peppersawce) [reply games-rpg/flare/flare-1.14.recipe:21]: 'cause it builds, runs and finds all the needed paths by itself even without the patch, so why keep it?

File: games-rpg/flare/additional-files/flare.rdef.in [hunk 2]
```diff
@@ -9,10 +9,10 @@ resource app_version {
 	internal = 0,
 
 	short_info = "Flare",
-	long_info = "@LONG_INFO@"
+	long_info = "Free/Libre Action Roleplaying Engine"
```
Comment (Begasus) [inline games-rpg/flare/additional-files/flare.rdef.in:12]: Changes in this file are not needed, those are done with sed magic in the recipe, if you feel the SUMMARY is wrong, adjust that in the recipe.

File: games-rpg/flare/flare-1.14.recipe (hunk 1 above)
Comment (Begasus) [reply games-rpg/flare/flare-1.14.recipe:21]: Agree here with @korli it's not because it builds/runs without it should be thrown away, @threedeyes added the Haiku specific files/patches for a reason.

File: games-rpg/flare/additional-files/flare.rdef.in (hunk 2 above)
Comment (Peppersawce) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: Is sed even needed if the string will always be the same either way? I don't see a point in using sed in cases like this

Comment (jmairboeck) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: The point is to reuse the value of `$SUMMARY` from the recipe and not have the same string in two places.

Comment (Peppersawce) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: I'm sorry, that makes no sense to me, you end up with the same string in two places either way. I guess it would be useful in case the project ever changes its name, but even if that was the case there's no guarantee one could reuse the summary in the same way, ie. maybe thy rename themselves to just "FLARE" and then it would be a bad string for "long_info" either way.

File: games-rpg/flare/flare-1.14.recipe (hunk 1 above)
Comment (Peppersawce) [reply games-rpg/flare/flare-1.14.recipe:21]: Alright, will wait for feedback from the patch's creator, there may be more to that patch than I realized.

File: games-rpg/flare/additional-files/flare.rdef.in (hunk 2 above)
Comment (Begasus) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: SUMMARY has to "always" be at least 3 words (guidelines), these sed things and rdef's have been in use to ease the updates on future versions, nothing wrong with that, if upstream changes it's name or the like, the recipe should also reflect that, it's like "libVersion" you could drop that, but that would mean on a next update you would have to change all instances where libVersion is called from/declared in.

Comment (Peppersawce) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: To me it makes sense for version numbers and strings that are meant to change from version to version, long info and summary would stay the same regardless of recipe updates and if they change there's no actual guarantee the sed thing would still apply, imo it is just an extra command in the recipe to change a string that is very unlikely to ever change.

Comment (Begasus) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: I fail to see why the sed thing would fail to apply?

Comment (Peppersawce) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: I fail to see why we need an extra command to change a string that's staying the same regardless

Comment (Peppersawce) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: I get that, I'm not willing to back down either. :)
Sorry guys, I don't see any actual benefit of using variables for fields like that, simple as.

Comment (korli) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: Just imagine the rdef.in is a template you can't edit, because it's the same for every recipe. We might actually drop the rdef.in in favor of recipe scriptlet in the future.

Comment (Peppersawce) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: It's the same for every recipe you say... to me that's a confusing statement, this is the first rdef.in I've seen with so many variables in it, most only have version numbers as variables and that's it. Version numbers come and go, sure, but is there any point in having a variable that should never change? That's my issue with it, fields like long_info and app_signature realistically have little reason to change overtime, unless a project shifts focus entirely.

Comment (korli) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: So will you update every rdef.in in the repo because they are confusing to you?

Comment (Peppersawce) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: You're saying every rdef.in in this repo has app_signature as a variable then?

**Comment by Peppersawce** [comment]
Closing this request as I don't feel like spending 2 days straight arguing about semantics, it's a waste of my time and of your time. If anybody else wants to update the old recipe it takes literally 2 minutes.

**Comment by korli** [comment]
Well the more you changes the more the review is difficult and long.

File: games-rpg/flare/additional-files/flare.rdef.in (hunk 2 above)
Comment (jmairboeck) [reply games-rpg/flare/additional-files/flare.rdef.in:12]: Just for reference: there is a counter-example that I wrote: https://github.com/haikuports/haikuports/commit/b3510feb4985b2791f8f31dbfcf3950025213fdd

The reason there is that perl module versions (which is used as `$portVersion`) are weird and aren't equal to the actual version of the app (perl module versions are single floating point numbers, not usual version numbers). So in the end I didn't even bother with `.rdef.in` files and instead just used `.rdef` files and copied them directly. But, admittedly, that is a special case-
